src/common/dsp/SurgeVoice.cpp

```diff
@@ -330,15 +361,40 @@ void SurgeVoice::uber_release()
 
 void SurgeVoice::update_portamento()
 {
-   // float ratemult=1;
-   // if(zone->portamento_mode) ratemult = 12/(0.00001+fabs(key-portasrc_key));
+   float const_rate_factor = 1.0;
+   int quantStep = 12;
+   
+   if (!storage->isStandardTuning && storage->currentScale.count > 1)
+      quantStep = storage->currentScale.count;
+
+   // portamento constant rate mode (multiply portamento time with every octave traversed (or scale length in case of microtuning)
+   if (scene->portamento.porta_constrate)
+      const_rate_factor = (1.f / ((1.f / quantStep) * fabs(state.getPitch() - state.portasrc_key))) + 0.00000001;
```

Comment:
If this +0.0000001 is supposed to suppress divide by zero then it is in the wrong place in the ()s. Also it is too small for a float. You want 0.00001 more likely.

Reply:
I see it now too, thanks!